query-features-with-arcade-expression/src/main/java/com/esri/arcgismaps/sample/queryfeatureswitharcadeexpression/MainActivity.kt

```diff
@@ -0,0 +1,210 @@
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val infoTextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // progress indicator
+    private val progressBar by lazy {
+        activityMainBinding.progressBar
+    }
+
+    // setup the red pin marker image as a bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(resources, R.drawable.map_pin_symbol)
+        BitmapDrawable(resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        val markerSymbol = PictureMarkerSymbol(markerDrawable).apply {
+            // resize the symbol into a smaller size
+            width = 30f
+            height = 30f
+            // offset in +y axis so the marker spawned is right on the touch point
+            offsetY = 25f
+        }
+        // create the graphic from the symbol
+        Graphic(symbol = markerSymbol)
+    }
+
+    // create a graphic overlay
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create a portal item with the itemId of the web map
+        val portal = Portal("https://www.arcgis.com/")
+        val portalItem = PortalItem(portal, "539d93de54c7422f88f69bfac2aebf7d")
+        // create and add a map with with portal item
+        val map = ArcGISMap(portalItem)
+        // add the marker graphic to the graphics overlay
+        graphicsOverlay.graphics.add(markerGraphic)
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        lifecycleScope.launch {
+            // show an error and return if the map load failed
+            map.load().onFailure {
+                return@launch showError("Error loading map:${it.message}")
+            }
+
+            // get the RPD Beats layer from the map's operational layers
+            val policeBeatsLayer =
+                map.operationalLayers.firstOrNull { layer ->
+                    layer.id == "RPD_Reorg_9254"
+                } ?: return@launch showError("Error finding RPD Beats layer")
+
+            // capture and collect when the user taps on the screen
+            mapView.onSingleTapConfirmed.collect { event ->
+                // update the marker location to where the user tapped on the map
+                markerGraphic.geometry = event.mapPoint
```

Comment:
```suggestion
                markerGraphic.geometry = event.mapPoint
                // center viewpoint to the tapped point
                event.mapPoint?.let { mapView.setViewpointCenter(it) }
```
Probably would be helpful to center the viewpoint to the tapped point.